ick/cmdline.py

```diff
@@ -110,17 +112,27 @@ def run(ctx, dry_run: bool, patch: bool, yolo: bool, filters: list[str]):
 
     # DO THE NEEDFUL
 
+    rule_modified = {}
+
     r = Runner(ctx.obj, ctx.obj.repo, explicit_project=None)
     for result in r.run():
-        print(f"-> [bold]{result.rule}[/bold] on {result.project}", end="")
-        if result.finished.error:
+        if not json_flag:
+            print(f"-> [bold]{result.rule}[/bold] on {result.project}", end="")
+        if result.finished.error and not json_flag:
             print("[red]ERROR[/red]")
             for line in result.finished.message.splitlines():
                 print("    ", line)
         else:
-            print("[green]OK[/green]")
+            if not json_flag:
+                print("[green]OK[/green]")
+
+        if json_flag:
+            ok_status = not result.finished.error
+            modified = rule_modified.get(result.rule, False) or len(result.modifications) > 0
```

Comment:
if the filename and diffstat might be useful to you, consider making this be a list of (filename, diffstat) or None in the case of no modifications.  That's more future proof than just a bool.